feat: add attributes for drag region (#6362)
tauri-apps/tauri: .changes/add-attribute-for-drag-region.md

---
"tauri": "minor"
---

Added new drag attributes,`data-tauri-drag-region-container`, `data-tauri-drag-region-titlebar`, `data-tauri-drag-region-interactive`, to enable drag on children, control maximize on double click, control interactive elements' behavior.

Description:

`data-tauri-drag-region`: mark this element as "drag region"

`data-tauri-drag-region-container`: children are also considered drag regions

`data-tauri-drag-region-titlebar`: the "drag region" maximizes window on double click (default when only `data-tauri-drag-region` is used)

`data-tauri-drag-region-exclude`: this element doesn't trigger `drag`

Example:

```html

<div>
  <div class="title"
       data-tauri-drag-region-container="true"
       data-tauri-drag-region-titlebar="true"
  >
    <div>Title</div>
    <div>Some decoration</div>
    <button data-tauri-drag-region-exclude>close</button>
  </div>
  <div class="content"
       data-tauri-drag-region-container="true"
  >
    <div>content</div>
    <div>content</div>
  </div>

> </div>

Reviewer: ```suggestion
<div class="titlebar" data-tauri-drag-region-container data-tauri-drag-region-titlebar>
	<span>Title</span>
	<span>Some decoration</span>
    <button  data-tauri-drag-region-exclude>Close</button>
</div>
```

---
"tauri": "minor"
---

Added
attribute `data-tauri-drag-region-container`, `data-tauri-drag-region-title`, `data-tauri-drag-region-interactive`, to

> enable drag on children, control maximize on double click, control is or not interactable on button/input/etc.

Reviewer: ```suggestion
Added new drag attributes,`data-tauri-drag-region-container`, `data-tauri-drag-region-title`, `data-tauri-drag-region-interactive`, to enable drag on children, control maximize on double click, control interactive elements behavior.
```

Description:

`data-tauri-drag-region`: mark this element as "drag region"

> `data-tauri-drag-region-container`: mark the "drag region" effects its children

Reviewer: ```suggestion
`data-tauri-drag-region-container`: children are also considered drag regions
```

> `data-tauri-drag-region-title`: mark the "drag region" maximize on double click (default add when

Reviewer: Should probably be called `data-tauri-drag-region-titlebar`

> `data-tauri-drag-region-title`: mark the "drag region" maximize on double click (default add when
> only `data-tauri-drag-region` is set)

Reviewer: ```suggestion
`data-tauri-drag-region-title`: the "drag region" maximizes window on double click (default when only `data-tauri-drag-region` is used)
```

`data-tauri-drag-region-interactive`: mark the "drag region" wouldn't prevent event to its children that interact-able,

> like button, input, etc. (check by "is property 'value' exists")

Reviewer: The behavior should be changed to be used on the element that needs to be excluded.
```suggestion
`data-tauri-drag-region-exclude`: this element doesn't trigger `drag`
```

Author: In my design, exclude single element is done by set `data-tauri-drag-region` to false,
and if `data-tauri-drag-region-container` is also used, then also exclude it's children

the `data-tauri-drag-region-interactive` is for mark a container that shouldn't prevent events to default interact-able element.

Reviewer 2: yeah I understand but this feels quite unpredictable for me, I'd rather use an explicit attribute to mark an element as not draggable instead of adding to the parent.

Reviewer: yeah I understand but this feels quite unpredictable for me, I'd rather use an explicit attribute to mark an element as not draggable instead of adding to the parent.
